gstcefsrc.cc

```diff
@@ -778,13 +789,22 @@ gst_cef_src_start(GstBaseSrc *base_src)
   GST_OBJECT_UNLOCK (src);
 
   browserClient = new BrowserClient(renderHandler, audioHandler, requestHandler, displayHandler, src);
-  CefPostTask(TID_UI, base::BindOnce(&BrowserClient::MakeBrowser, browserClient.get(), 0));
+#ifdef __APPLE__
+  if (pthread_main_np()) {
```

Comment:
A similar change should be needed in `stop()` then? You'd close the browser there and wait for it to be closed, but if that is called from the main thread this would never happen because the closing would happen from that very thread's event loop?

Reply:
Yeah, that's a correct insight. Calls to CEF can be posted from either the main thread or the creator's thread, but the wait must happen only if the threads are different.